alibi_detect/cd/sklearn/classifier.py

```diff
@@ -0,0 +1,234 @@
+            or via reservoir sampling with size n. For the former, the parameter equals {'last': n} while
+            for reservoir sampling {'reservoir_sampling': n} is passed.
+        preprocess_fn
+            Function to preprocess the data before computing the data drift metrics.
+        preds_type
+            Whether the model outputs 'probs' or 'scores'.
+        binarize_preds
+            Whether to test for discrepancy on soft (e.g. probs/scores) model predictions directly
+            with a K-S test or binarise to 0-1 prediction errors and apply a binomial test.
+        train_size
+            Optional fraction (float between 0 and 1) of the dataset used to train the classifier.
+            The drift is detected on `1 - train_size`. Cannot be used in combination with `n_folds`.
+        n_folds
+            Optional number of stratified folds used for training. The model preds are then calculated
+            on all the out-of-fold predictions. This allows to leverage all the reference and test data
+            for drift detection at the expense of longer computation. If both `train_size` and `n_folds`
+            are specified, `n_folds` is prioritized.
+        retrain_from_scratch
+            Whether the classifier should be retrained from scratch for each set of test data or whether
+            it should instead continue training from where it left off on the previous set.
+        seed
+            Optional random seed for fold selection.
+        use_calibration
+            Whether to use calibration. Whether to use calibration. Calibration can be used on top of any model.
+        calibration_kwargs
+            Optional additional kwargs for calibration.
+            See https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html
+            for more details.
+        data_type
+            Optionally specify the data type (tabular, image or time-series). Added to metadata.
+        """
+        super().__init__(
+            x_ref=x_ref,
+            p_val=p_val,
+            preprocess_x_ref=preprocess_x_ref,
+            update_x_ref=update_x_ref,
+            preprocess_fn=preprocess_fn,
+            preds_type=preds_type,
+            binarize_preds=binarize_preds,
+            train_size=train_size,
+            n_folds=n_folds,
+            retrain_from_scratch=retrain_from_scratch,
+            seed=seed,
+            data_type=data_type
+        )
+
+        if preds_type not in ['probs', 'scores']:
+            raise ValueError("'preds_type' should be 'probs' or 'scores'")
+
+        self.meta.update({'backend': 'sklearn'})
+        self.original_model = model
+        self.use_calibration = use_calibration
+        self.calibration_kwargs = dict() if calibration_kwargs is None else calibration_kwargs
+        self.model = self._clone_model()  # type: ClassifierMixin
+
+    def _has_predict_proba(self, model) -> bool:
```

Comment:
Should this function be somewhere in `sklearn.utils` instead? @jklaise

Reply:
If it becomes reused for other things then that would make sense. I was going to suggest factoring this out as a stand-alone function to begin with, but it's a bit inconvenient as it relies on a dataset (ideally, this would be a single-argument function with `model`, but @RobertSamoilescu tried using `inspect` which didn't seem to give the desired results on the latest `sklearn`, so the prediction call is still in place). I would say it's fine right now as it's the only place we use it and better to have it defined in the same module.

Reply:
Ok sounds good!